includes/functions/functions_email.php

```diff
@@ -783,3 +783,19 @@ function zen_get_email_from_customers_id($customers_id) {
     if ($customers_values->EOF) return '';
     return $customers_values->fields['customers_email_address'];
   }
+
+
+  function zen_db_prepare_input_html_safe($string) {
+    if (is_string($string)) {
+      return trim(stripslashes($string));
+    } elseif (is_array($string)) {
+      reset($string);
+      while (list($key, $value) = each($string)) {
```

Comment:
Note: This `reset/while/list/each` syntax is deprecated since PHP 7.2

Related: https://github.com/zencart/zencart/pull/1377

Reply:
Good catch.  Thanks!